src/wallet/wallet.h

```diff
@@ -578,20 +578,30 @@ class COutput
      */
     bool fSafe;
 
+    const libmw::Coin *mwCoin;
```

Comment:
Changing assumptions for CInputCoin is probably fine, since it's limited exclusively to coin selection. I really don't like the idea of changing the assumptions around COutput though if we can avoid it. By opening up the opportunity for things like COutput.tx to be null, we're just begging for memory issues imho.

We'll chat in Telegram about some other options for this.